packages/core/src/Serialization/TransactionBody/TransactionBody.ts

```diff
@@ -506,31 +438,33 @@ export class TransactionBody {
    */
   static fromCore(coreTransactionBody: Cardano.TxBody): TransactionBody {
     const body = new TransactionBody(
-      coreTransactionBody.inputs.map((input) => TransactionInput.fromCore(input)),
+      new CborSet<TransactionInput>(coreTransactionBody.inputs.map((input) => TransactionInput.fromCore(input))),
```

Comment:
Why not implement in the CborSet class a fromCore and toCore, so you can abstract away this convertions

Reply:
Done [here](https://github.com/input-output-hk/cardano-js-sdk/commit/eb316344a94eba2ca9b8fba73c1e78198e47da8b)